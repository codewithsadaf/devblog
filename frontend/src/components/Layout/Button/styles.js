//imports
import styled, { css } from "styled-components";

const handleButtonVariant = {
    default: (theme) => css`
        background: ${theme.colors.primaryColor};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: transparent;
            color: ${theme.colors.primaryColor};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,
    outline: (theme) => css`
        background: transparent;
        color: ${theme.colors.primaryColor};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: ${theme.colors.primaryColor};
            color: ${theme.colors.white};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,

    success: (theme) => css`
        background: ${theme.colors.success};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.success};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,

    danger: (theme) => css`
        background: ${theme.colors.danger};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.danger};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,
};

const handleUppercase = () => css`
    text-transform: uppercase;
`;

const handleMaxWidth = () => css`
    width: 100%;
    padding: 1em 0;
`;

const handleSmallStyles = (theme) => css`
    padding: 7px 30px;
    border-radius: 4px;
    font-size: ${theme.font.sizes.small};
    border: 1px solid;

    &:hover {
        border: 1px solid;
    }
`;

export const ButtonContainer = styled.button`
    ${({ theme, buttonVariant, uppercase, maxWidth, isSmall }) => css`
        padding: 0.5em 1.8em;
        font-size: 1em;
        border: none;
        font-weight: bold;
        cursor: pointer;
        ${handleButtonVariant[buttonVariant](theme)};
        ${uppercase && handleUppercase()};
        ${maxWidth && handleMaxWidth()};
        ${isSmall && handleSmallStyles(theme)};
    `}
`;
