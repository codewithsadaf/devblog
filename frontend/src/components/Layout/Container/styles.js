//imports
import styled from "styled-components";

export const Container = styled.div`
    min-height: 60vh;
    margin-bottom: 5em;
    padding: 0 2em;
`;
