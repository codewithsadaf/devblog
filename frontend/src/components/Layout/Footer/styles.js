//imports
import styled, { css } from "styled-components";

export const FooterWrapper = styled.footer`
    ${({ theme }) => css`
        height: 250px;
        width: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        background-color: ${theme.colors.footerBackground};
        text-align: center;
    `}
`;
export const FooterHeading = styled.h3`
    ${({ theme }) => css`
        @media ${theme.media.lteMedium} {
            font-size: ${theme.font.sizes.medium};
            line-height: ${theme.font.line_height.h6};
        }
    `}
`;

export const CopyrightText = styled.span`
    display: block;
    margin-top: 0.8em;
`;
