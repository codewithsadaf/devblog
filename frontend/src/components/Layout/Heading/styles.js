//imports
import styled, { css } from "styled-components";

const headingSize = {
    huge: (theme) => css`
        font-size: ${theme.font.sizes.xhuge};
        line-height: ${theme.font.line_height.h1};
        ${mediaFont(theme)};
    `,
    big: (theme) => css`
        font-size: ${theme.font.sizes.huge};
        line-height: ${theme.font.line_height.h2};
    `,
    large: (theme) => css`
        font-size: ${theme.font.sizes.xxlarge};
        line-height: ${theme.font.line_height.h3};
    `,
    medium: (theme) => css`
        font-size: ${theme.font.sizes.medium};
        line-height: ${theme.font.line_height.h6};
    `,
    small: (theme) => css`
        font-size: ${theme.font.sizes.small};
        line-height: ${theme.font.line_height.small};
    `,
    xsmall: (theme) => css`
        font-size: ${theme.font.sizes.xsmall};
        line-height: ${theme.font.line_height.small};
    `,
};

const mediaFont = (theme) => css`
    @media ${theme.media.lteMedium} {
        font-size: ${theme.font.sizes.xlarge};
        line-height: ${theme.font.line_height.h4};
    }
`;

export const Container = styled.h1`
    ${({ theme, size, uppercase, bold }) => css`
        text-transform: ${uppercase ? "uppercase" : "none"};
        font-weight: ${bold ? "bold" : 500};
        ${headingSize[size](theme)};
    `}
`;
