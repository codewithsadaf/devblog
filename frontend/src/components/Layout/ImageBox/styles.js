//imports
import styled, { css } from "styled-components";

const handleBigStyle = () => css`
    width: 100%;
    height: auto;
`;

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const ImageElm = styled.img`
    ${({ rounded }) => css`
        width: 150px;
        height: 150px;
        object-fit: cover;
        display: block;
        margin: 1em auto;
        border-radius: ${rounded ? "50%" : "7px"};
    `}
`;
