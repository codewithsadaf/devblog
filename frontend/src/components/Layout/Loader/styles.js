//imports
import styled, { css } from "styled-components";

export const Container = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100vh;
    background: rgba(0, 0, 0, 0.6);
    z-index: 99999;
    pointer-events: all;
`;

export const Circle = styled.div`
    ${({ theme }) => css`
        &,
        &::before,
        &::after {
            border-radius: 50%;
            width: 2.5em;
            height: 2.5em;
            animation-fill-mode: both;
            animation: bblFadInOut 1.8s infinite ease-in-out;
        }

        & {
            color: ${theme.colors.white};
            font-size: 7px;
            position: relative;
            text-indent: -9999em;
            transform: translateZ(0);
            animation-delay: -0.16s;
        }

        &::before,
        &:after {
            content: "";
            position: absolute;
            top: 0;
        }

        &::before {
            left: -3.5em;
            animation-delay: -0.32s;
        }

        &::after {
            left: 3.5em;
        }

        @keyframes bblFadInOut {
            0%,
            80%,
            100% {
                box-shadow: 0 2.5em 0 -1.3em;
            }
            40% {
                box-shadow: 0 2.5em 0 0;
            }
        }
    `}
`;
