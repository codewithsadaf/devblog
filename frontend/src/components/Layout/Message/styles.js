//imports
import styled, { css } from "styled-components";
//react icons
import { IoClose } from "react-icons/io5";

export const Container = styled.div`
    ${({ theme }) => css`
        display: flex;
        align-items: center;
        justify-content: space-between;
        position: fixed;
        bottom: 30px;
        left: 50%;
        transform: translateX(-50%);
        padding: 25px 30px;
        min-width: 25%;
        border-radius: 7px;
        z-index: 9999;
        font-weight: bold;
        box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
        transition: ${theme.transition.tran_03};

        &.success {
            background-color: ${theme.colors.success};
            color: ${theme.colors.white};
        }

        &.error {
            background-color: ${theme.colors.danger};
            color: ${theme.colors.white};
        }

        @media ${theme.media.lteMedium} {
            bottom: 0;
            right: 50%;
            left: 50%;
            min-width: 90%;
            transform: translate(-50%, -50%);
        }
    `}
`;

export const CloseIcon = styled(IoClose)`
    ${({ theme }) => css`
        color: ${theme.colors.white};
        cursor: pointer;
        font-size: 19px;
        margin-left: 0.3em;
    `}
`;

export const MessageText = styled.span`
    display: block;
`;
