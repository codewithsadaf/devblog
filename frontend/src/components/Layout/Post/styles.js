//imports
import styled, { css } from "styled-components";
//components styles
import { Text as TextComponent } from "../TextComponent/styles";
import { Container as Heading } from "../Heading/styles";

export const Container = styled.div`
    ${({ theme }) => css`
        width: 100%;
        margin: 0 auto 3em;
        max-width: 600px;

        ${TextComponent} {
            color: #444;
            font-size: ${theme.font.sizes.small};
            line-height: ${theme.font.line_height.small};
            font-style: italic;
        }

        @media ${theme.media.lteMedium} {
            ${Heading} {
                font-size: ${theme.font.sizes.xlarge};
                line-height: ${theme.font.line_height.h4};
            }
        }
    `}
`;

export const Image = styled.img`
    display: flex;
    width: 100%;
    margin-bottom: 1em;
    border-radius: 4px;
`;

export const TagsWrapper = styled.div`
    display: flex;
    justify-content: flex-start;
    align-items: center;
    margin: 0.8em 0;
    flex-wrap: wrap;
`;

export const TagText = styled.span`
    display: block;
    margin-right: 0.5em;
    font-weight: 500;

    &::before {
        content: "#";
        font-weight: bold;
    }
`;
