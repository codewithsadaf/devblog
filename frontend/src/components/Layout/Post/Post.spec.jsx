//imports
import { fireEvent, screen, waitFor } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
//styles
import { renderTheme } from "../../../styles/render-theme";
//components
import { Post } from "./";

describe("<Post />", () => {
    describe("Rendering", () => {
        beforeEach(() => {
            renderTheme(
                <BrowserRouter>
                    <Post
                        imgSrc={"image.png"}
                        headingText={"heading example"}
                        author={"author example"}
                        tags={["#test1", "#test2"]}
                        postData={"2023-04-05T19:27:03.125+00:00"}
                        id="1"
                    />
                </BrowserRouter>
            );
        });

        it("should render Post", () => {
            const heading = screen.getByRole("heading", {
                name: "heading example",
            });

            expect(heading).toBeInTheDocument();
        });

        it("should render Post with correct alt value", () => {
            const postImage = document.querySelector("img");

            expect(postImage.alt).toContain("heading example");
        });

        it("should render Post with correct src value", () => {
            const postImage = document.querySelector("img");

            expect(postImage.src).toContain("image.png");
        });

        it("should render Post author with correct name and data", () => {
            // Regex to include optional whitespace around characters.
            const authorText = screen.getByText(
                /Por:\s*author example\s*-\s*5 de Abril 2023/
            );

            expect(authorText).toBeInTheDocument();
        });

        it("should render each post tag of array", () => {
            expect(screen.getByText("test1")).toBeInTheDocument();
            expect(screen.getByText("test2")).toBeInTheDocument();
        });

        test("should render TagText with a before style rule", () => {
            const tagText = screen.getByText("test1");

            expect(tagText).toHaveStyleRule("content", '"#"', {
                modifier: "&::before",
            });
            expect(tagText).toHaveStyleRule("font-weight", "bold", {
                modifier: "&::before",
            });
        });
    });

    describe("Functionality", () => {
        it("should call function on button click to redirect to post page", () => {
            //function mock
            const navigateMock = jest.fn();
            /* doMock method to replace "react-router-dom" module and then 
            replace useNavigate hook to navigateMock */
            jest.doMock("react-router-dom", () => ({
                ...jest.requireActual("react-router-dom"),
                useNavigate: () => navigateMock,
            }));

            renderTheme(
                <BrowserRouter>
                    <Post
                        imgSrc={"image.png"}
                        headingText={"heading example"}
                        author={"author example"}
                        tags={["#test1", "#test2"]}
                        postData={"2023-04-05T19:27:03.125+00:00"}
                        id="1"
                    />
                </BrowserRouter>
            );

            const buttonElm = screen.getByRole("button", { name: "Ler" });

            fireEvent.click(buttonElm);

            // waitFor to wait navigateMock be called
            waitFor(() => {
                expect(navigateMock).toHaveBeenCalledWith("/post/1");
            });
        });
    });

    describe("Snapshot", () => {
        it("should match snapshot", () => {
            const { container } = renderTheme(
                <BrowserRouter>
                    <Post
                        imgSrc={"image.png"}
                        headingText={"heading example"}
                        author={"author example"}
                        tags={["#test1", "#test2"]}
                        postData={"2023-04-05T19:27:03.125+00:00"}
                        id="1"
                    />
                </BrowserRouter>
            );

            expect(container.firstChild).toMatchSnapshot();
        });
    });
});
