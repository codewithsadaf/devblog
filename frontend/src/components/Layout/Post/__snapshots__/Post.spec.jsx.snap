// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> Snapshot should match snapshot 1`] = `
.c5 {
  font-size: 1em;
  line-height: 150%;
}

.c3 {
  text-transform: none;
  font-weight: bold;
  font-size: 2em;
  line-height: 110%;
}

.c0 {
  width: 100%;
  margin: 0 auto 3em;
  max-width: 600px;
}

.c0 .c4 {
  color: #444;
  font-size: 0.8em;
  line-height: 150%;
  font-style: italic;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin-bottom: 1em;
  border-radius: 4px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.8em 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c7 {
  display: block;
  margin-right: 0.5em;
  font-weight: 500;
}

.c7::before {
  content: "#";
  font-weight: bold;
}

.c8 {
  padding: 0.5em 1.8em;
  font-size: 1em;
  border: none;
  font-weight: bold;
  cursor: pointer;
  background: transparent;
  color: #000;
  border: 2px solid #000;
  border-radius: 4px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  background: #000;
  color: #fff;
  border: 2px solid #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

@media (max-width:768px) {
  .c0 .c2 {
    font-size: 1.7em;
    line-height: 120%;
  }
}

<div
  class="c0"
>
  <img
    alt="heading example"
    class="c1"
    src="image.png"
  />
  <h3
    class="c2 c3"
  >
    heading example
  </h3>
  <p
    class="c4 c5"
  >
    Por: 
    author example
     - 
    5 de Abril 2023
  </p>
  <div
    class="c6"
  >
    <span
      class="c7"
    >
      test1
    </span>
    <span
      class="c7"
    >
      test2
    </span>
  </div>
  <button
    class="c8"
    type="button"
  >
    Ler
  </button>
</div>
`;
